tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,340 @@
+    )
+
+    if chr_place == 23:
+        filter_nonpar_expr = sex_mt.locus.in_y_nonpar()
+    elif chr_place == 22:
+        filter_nonpar_expr = sex_mt.locus.in_x_nonpar()
+
+    if chr_place in [22, 23]:
+        logger.info("Filtering to non-PAR regions")
+        sex_mt = sex_mt.filter_rows((filter_nonpar_expr), keep=True)
+
+    # Filter to common SNVs above defined callrate (should only have one index in the array because the MT only contains biallelic variants)
+    sex_mt = sex_mt.filter_rows(sex_mt[af_field] > af_threshold)
+
+    # TODO: Make callrate filtering optional before adding code to gnomad_methods
+    sex_mt = hl.variant_qc(sex_mt)
+    sex_mt = sex_mt.filter_rows(sex_mt.variant_qc.call_rate > call_rate_threshold)
+
+    logger.info("Returning mean coverage on chromosome %s...", chr_name)
+
+    sex_mt = sex_mt.annotate_cols(**{f"{chr_name}_mean_dp": hl.agg.mean(sex_mt.DP)})
+    sex_ht = sex_mt.cols()
+    return sex_ht
+
+
+def run_hails_impute_sex(
+    mt: hl.MatrixTable,
+    build: str,
+    outdir: str,
+    callset_name: str,
+    xy_fstat_threshold: float = 0.75,
+    xx_fstat_threshold: float = 0.5,
+    aaf_threshold: float = 0.05,
+) -> hl.Table:
+    """
+    Impute sex, annotate MatrixTable with results, and output a histogram of fstat values.
+
+    :param MatrixTable mt: MatrixTable containing samples to be ascertained for sex
+    :param build: Reference used, either GRCh37 or GRCh38
+    :param callset_name: Basename for callset and output results
+    :param outdir: Directory to output results
+    :param xy_fstat_threshold: F-stat threshold above which a sample will be called XY. Default is 0.75
+    :param xx_fstat_threshold: F-stat threshold below which a sample will be called XX. Default is 0.5
+    :param aaf_threshold: Alternate allele frequency threshold for `hl.impute_sex`. Default is 0.05
+    :return: Table with imputed sex annotations
+    """
+
+    logger.warning(
+        "User needs to decide on fstat thresholds for male/female by looking at fstat plots"
+    )
+
+    # Filter to the X chromosome and impute sex
+    mt = hl.filter_intervals(
+        mt,
+        [
+            hl.parse_locus_interval(
+                get_reference_genome(mt.locus).x_contigs[0], reference_genome=build
+            )
+        ],
+    )
+    sex_ht = hl.impute_sex(
+        mt.GT,
+        aaf_threshold=aaf_threshold,
+        male_threshold=xy_fstat_threshold,
+        female_threshold=xx_fstat_threshold,
+    )
+    mt = mt.annotate_cols(**sex_ht[mt.col_key])
+    sex_ht = mt.cols()
+
+    # Plot histogram of fstat values
+    df = sex_ht.to_pandas()
+    plt.clf()
+    plt.hist(df["f_stat"])
+    plt.xlabel("Fstat")
+    plt.ylabel("Frequency")
+    plt.axvline(xy_fstat_threshold, color="blue", linestyle="dashed", linewidth=1)
+    plt.axvline(xx_fstat_threshold, color="red", linestyle="dashed", linewidth=1)
+
+    outplot = f"{outdir}/fstat_{callset_name}.png"
```

Comment:
```suggestion
    # TODO: Change this path to be an argument (rather than having code decide on the output path)
    outplot = f"{outdir}/fstat_{callset_name}.png"
```